graphcanvas/layout.py

```diff
@@ -12,7 +12,9 @@ def tree_layout(graph, dim=2, scale=1):
     if not networkx.is_directed_acyclic_graph(graph):
         raise ValueError('graph must not contain cycles')
 
-    roots = numpy.array(graph.nodes())[numpy.where(numpy.array(graph.in_degree()) < 1)]
+    roots = numpy.array([
```

Comment:
`np.array()` does not consume generators the way you intend it to. You have to build a concrete sequence.